crates/sui-gateway/src/event_api.rs

```diff
@@ -0,0 +1,96 @@
+use std::collections::BTreeMap;
+use std::sync::Arc;
+
+use jsonrpsee_core::error::SubscriptionClosed;
+use jsonrpsee_core::server::rpc_module::PendingSubscription;
+use jsonrpsee_proc_macros::rpc;
+use serde::Deserialize;
+use serde::Serialize;
+use strum::IntoEnumIterator;
+use strum_macros::EnumIter;
+use tokio::sync::broadcast;
+use tokio::sync::broadcast::Sender;
+use tokio_stream::wrappers::BroadcastStream;
+use tokio_stream::StreamExt;
+use tracing::{debug, warn};
+
+use sui_core::gateway_types::SuiEvent;
+
+#[rpc(server, client, namespace = "sui")]
+pub trait EventApi {
+    #[subscription(name = "subEvent", unsubscribe = "unsubEvent", item = SuiEvent)]
```

Comment:
is `subEvent` going to be the external JsonRPC method name?   Why not use the full name `subscribeEvents` or something like that?

Also, I believe the ask was to subscribe by either event type or Move package/module.